src/gsy_e/models/strategy/future/strategy.py

```diff
@@ -48,6 +47,13 @@ def _get_all_time_slots(area: "Area") -> List[DateTime]:
             return []
         return area.future_markets.market_time_slots
 
+    def time_for_price_update(self, strategy: "BaseStrategy", time_slot: DateTime) -> bool:
+        """Check if the prices of bids/offers should be updated."""
+        return (
+                self._elapsed_seconds(strategy.area)
+                - self.market_slot_added_time_mapping[time_slot] >= (
+                        self.update_interval.seconds * self.update_counter[time_slot]))
+
```

Comment:
If the elapsed seconds = 3000 seconds and the slot is created at 1000 seconds, then the elapsed seconds = 3000 - 1000 = 2000 seconds (age of the market)

Reply:
OK, this makes sense now. It took me a while to understand. Thanks a lot for the explanation (also the verbose one!)